File: app/src/main/java/com/arielfaridja/ezrahi/entities/ReportType.java
package com.arielfaridja.ezrahi.entities;

public enum ReportType {
    GENERAL(0),
    MEDICAL(1);

    private final int value;

    ReportType(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }
}

File: app/src/main/java/com/arielfaridja/ezrahi/entities/ReportType.kt
package com.arielfaridja.ezrahi.entities

enum class ReportType(val value: Int) {
    GENERAL(0),
    MEDICAL(1),
    UNKNOWN(-1);

    companion object {
        fun getByValue(value: Int): ReportType = when (value) {
            0 -> GENERAL
            1 -> MEDICAL
            else -> UNKNOWN
        }
    }
}
